src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/MeasurementExtractor.cs

```diff
@@ -0,0 +1,165 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using EnsureThat;
+using Microsoft.Health.Fhir.Ingest.Data;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class MeasurementExtractor : IContentTemplate
+    {
+        public const string MatchedToken = "matchedToken";
+
+        public MeasurementExtractor(
```

Comment:
It would be good to take this opportunity to inject an ILogger here for better traceability.

Reply:
I don't have any specific logs/telemetry to produce here at the moment. But I've added the `iLogger` as a member of the factory that creates this class, so if we decide on telemetry here it'll be trivial to pass an `iLogger` in.